File: JAVA/final-BT-java/src/models/Categorie.java
package models;

public class Categorie {

    private int idCategorie;
    private String nom;
    private TypeCategorie type;
    private int idUtilisateur;
    private boolean estCommune;

    public Categorie(int idCategorie, String nom, TypeCategorie type, int idUtilisateur, boolean estCommune) {
        this.idCategorie = idCategorie;
        this.nom = nom;
        this.type = type;
        this.idUtilisateur = idUtilisateur;
        this.estCommune = estCommune;
    }

    public Categorie() {
    }

    public int getIdCategorie() {
        return idCategorie;
    }

    public void setIdCategorie(int idCategorie) {
        this.idCategorie = idCategorie;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public TypeCategorie getType() {
        return type;
    }

    public void setType(TypeCategorie type) {
        this.type = type;
    }

    public int getIdUtilisateur() {
        return idUtilisateur;
    }

    public void setIdUtilisateur(int idUtilisateur) {
        this.idUtilisateur = idUtilisateur;
    }

    public boolean isEstCommune() {
        return estCommune;
    }

    public void setEstCommune(boolean estCommune) {
        this.estCommune = estCommune;
    }

    @Override
    public String toString() {
        return "Categorie [idCategorie=" + idCategorie + ", nom=" + nom + ", type=" + type + ", idUtilisateur="
                + idUtilisateur + ", estCommune=" + estCommune + "]";
    }

}

File: PHP/src/Entity/Categorie.php
<?php

namespace App\Entity;

class Categorie
{
    private ?int $idCategorie = null;
    private string $nom;
    private string $type;
    private bool $parDefaut;
    private int $idUtilisateur;

    public function getIdCategorie(): ?int
    {
        return $this->idCategorie;
    }

    public function setIdCategorie(?int $idCategorie): void
    {
        $this->idCategorie = $idCategorie;
    }

    public function getNom(): string
    {
        return $this->nom;
    }

    public function setNom(string $nom): void
    {
        $this->nom = $nom;
    }

    public function getType(): string
    {
        return $this->type;
    }

    public function setType(string $type): void
    {
        $this->type = $type;
    }

    public function isParDefaut(): bool
    {
        return $this->parDefaut;
    }

    public function setParDefaut(bool $parDefaut): void
    {
        $this->parDefaut = $parDefaut;
    }

    public function getIdUtilisateur(): int
    {
        return $this->idUtilisateur;
    }

    public function setIdUtilisateur(int $idUtilisateur): void
    {
        $this->idUtilisateur = $idUtilisateur;
    }
}